platform/CriticalSectionLock.h

```diff
@@ -34,35 +35,68 @@ namespace mbed {
   *     }
   *     // interrupts will be restored to their previous state
   * }
+  * void f() {
+  *     CriticalSectionLock cs(false);
+  *     // some code here
+  *
+  *     cs.lock();
+  *        // Code in this block will run with interrupts disabled
+  *     cs.unlock();
+  *     // interrupts will be restored to their previous state
+  *
+  *     cs.lock();
+  *        // Code in this block will run with interrupts disabled
+  *     cs.unlock();
+  *     // interrupts will be restored to their previous state
+  * }
   * @endcode
   */
 class CriticalSectionLock {
 public:
-    CriticalSectionLock() 
+    /** Create and initialize a CriticalSectionLock object.
+     * @param lock  critical section lock state after initialization
+     */
+    CriticalSectionLock(bool lock = true) : _locked(false)
     {
-        core_util_critical_section_enter();
+        if (lock) {
+            core_util_critical_section_enter();
```

Comment:
This change blurs the line even further between a regular lock and an RAII class so I would argue this is the wrong direction to be taking the code.